src/graphics/renderables/alert_manager.ts

```diff
@@ -0,0 +1,60 @@
+export enum AlertType {
+  Success = "alert-success",
+  Warning = "alert-warning",
+  Error = "alert-error",
+}
+
+export class AlertManager {
+  private static instance: AlertManager | null = null; // Singleton instance
+  private alertTimeout: NodeJS.Timeout | null = null;
+
+  public static getInstance(): AlertManager {
+    if (!AlertManager.instance) {
+      AlertManager.instance = new AlertManager();
+    }
+    return AlertManager.instance;
+  }
```

Comment:
This could be simplified to a module-level variable and a couple of functions:

```typescript
let alertTimeout: NodeJS.Timeout | null = null;

enum AlertType {
  Success = "alert-success",
  Warning = "alert-warning",
  Error = "alert-error",
}

function showAlert(message: string, type: string, duration = 3000): void {
    // ...
}

export function showSuccess(string message) {
    showAlert(message, AlertType.Success);
}

export function showError(string message) {
    showAlert(message, AlertType.Error);
}
```